exporter/awscloudwatchlogsexporter/config_test.go

```diff
@@ -142,6 +143,154 @@ func TestRetentionValidateWrong(t *testing.T) {
+		},
+	}
+	assert.NoError(t, component.ValidateConfig(cfg))
+
+}
+
+func TestTagsValidateTooManyTags(t *testing.T) {
+	m := make(map[string]*string)
+	avalue := "avalue"
+	for i := 0; i < 51; i++ {
+		m[strconv.Itoa(i)] = &avalue
+	}
+	defaultRetrySettings := exporterhelper.NewDefaultRetrySettings()
+	wrongcfg := &Config{
+		RetrySettings:      defaultRetrySettings,
+		LogGroupName:       "test-1",
+		LogStreamName:      "testing",
+		Endpoint:           "",
+		Tags:               m,
+		AWSSessionSettings: awsutil.CreateDefaultSessionConfig(),
+		QueueSettings: QueueSettings{
+			QueueSize: exporterhelper.NewDefaultQueueSettings().QueueSize,
+		},
+	}
+	assert.EqualError(t, component.ValidateConfig(wrongcfg), "invalid amount of items. Please input at most 50 tags")
+}
+
+func TestTagsValidateWrongKeyRegex(t *testing.T) {
+	avalue := "avalue"
+	defaultRetrySettings := exporterhelper.NewDefaultRetrySettings()
+	wrongcfg := &Config{
+		RetrySettings:      defaultRetrySettings,
+		LogGroupName:       "test-1",
+		LogStreamName:      "testing",
+		Endpoint:           "",
+		Tags:               map[string]*string{"***": &avalue},
+		AWSSessionSettings: awsutil.CreateDefaultSessionConfig(),
+		QueueSettings: QueueSettings{
+			QueueSize: exporterhelper.NewDefaultQueueSettings().QueueSize,
+		},
+	}
+	assert.EqualError(t, component.ValidateConfig(wrongcfg), "key - *** does not follow the regex pattern"+`^([\p{L}\p{Z}\p{N}_.:/=+\-@]+)$`)
+}
+
+func TestTagsValidateWrongValueRegex(t *testing.T) {
+	avalue := "***"
+	defaultRetrySettings := exporterhelper.NewDefaultRetrySettings()
+	wrongcfg := &Config{
+		RetrySettings:      defaultRetrySettings,
+		LogGroupName:       "test-1",
+		LogStreamName:      "testing",
+		Endpoint:           "",
+		Tags:               map[string]*string{"akey": &avalue},
+		AWSSessionSettings: awsutil.CreateDefaultSessionConfig(),
+		QueueSettings: QueueSettings{
+			QueueSize: exporterhelper.NewDefaultQueueSettings().QueueSize,
+		},
+	}
+	assert.EqualError(t, component.ValidateConfig(wrongcfg), "value - "+avalue+" does not follow the regex pattern"+`^([\p{L}\p{Z}\p{N}_.:/=+\-@]*)$`)
+}
+
+func TestTagsValidateKeyTooShort(t *testing.T) {
+	avalue := "avalue"
+	defaultRetrySettings := exporterhelper.NewDefaultRetrySettings()
+	wrongcfg := &Config{
+		RetrySettings:      defaultRetrySettings,
+		LogGroupName:       "test-1",
+		LogStreamName:      "testing",
+		Endpoint:           "",
+		Tags:               map[string]*string{"": &avalue},
+		AWSSessionSettings: awsutil.CreateDefaultSessionConfig(),
+		QueueSettings: QueueSettings{
+			QueueSize: exporterhelper.NewDefaultQueueSettings().QueueSize,
+		},
+	}
+	assert.EqualError(t, component.ValidateConfig(wrongcfg), "key -  has an invalid length. Please use keys with a length of 1 to 128 characters")
+}
+
+func TestTagsValidateKeyTooLong(t *testing.T) {
+	avalue := "avalue"
+	akey := ""
+	for i := 0; i < 129; i++ {
+		akey += "a"
+	}
```

Comment:
```suggestion
	akey := strings.Repeat("a", 129)
```

https://pkg.go.dev/strings#Repeat